doc/sphinx/language/core/variants.rst

```diff
@@ -64,6 +67,16 @@ be defined using :cmd:`Variant`.
       Variant option (A : Type) : Type := None : option A | Some : A -> option A.
       Variant sum (A B : Type) : Type := inl : A -> sum A B | inr : B -> sum A B.
 
+.. example::
+
+  :cmd:`Variant` cannot be used to define recursive types.
```

Comment:
Would be helpful to say how to define recursive types and perhaps hyperlink to the appropriate command.